js/Wikifier.js

```diff
@@ -134,104 +130,102 @@ Wikifier.prototype.subWikifyTerm = function(output,terminatorRegExp)
+	while(this.highlightMatch && (this.highlightRegExp.lastIndex > startPos) &&
+		  (this.highlightMatch.index < endPos) && (startPos < endPos)) {
 		//# Deal with any plain text before the highlight
 		if(this.highlightMatch.index > startPos) {
-			createTiddlyText(place,this.source.substring(startPos,this.highlightMatch.index));
+			createTiddlyText(place, this.source.substring(startPos, this.highlightMatch.index));
 			startPos = this.highlightMatch.index;
 		}
 		//# Deal with the highlight
-		var highlightEnd = Math.min(this.highlightRegExp.lastIndex,endPos);
-		createTiddlyElement(place,"span",null,"highlight",this.source.substring(startPos,highlightEnd));
+		var highlightEnd = Math.min(this.highlightRegExp.lastIndex, endPos);
+		createTiddlyElement(place, "span", null, "highlight", this.source.substring(startPos, highlightEnd));
 		startPos = highlightEnd;
 		//# Nudge along to the next highlight if we're done with this one
 		if(startPos >= this.highlightRegExp.lastIndex)
 			this.highlightMatch = this.highlightRegExp.exec(this.source);
 	}
 	//# Do the unhighlighted text left over
 	if(startPos < endPos) {
-		createTiddlyText(place,this.source.substring(startPos,endPos));
+		createTiddlyText(place, this.source.substring(startPos, endPos));
 	}
 };
 
-function wikify(source,output,highlightRegExp,tiddler)
+function wikify(source, output, highlightRegExp, tiddler)
 {
-	if(source) {
-		var wikifier = new Wikifier(source,getParser(tiddler),highlightRegExp,tiddler);
-		var t0 = new Date();
-		wikifier.subWikify(output);
-		if(tiddler && config.options.chkDisplayInstrumentation)
-			displayMessage("wikify:" +tiddler.title+ " in " + (new Date()-t0) + " ms");
-	}
+	if(!source) return;
+	var wikifier = new Wikifier(source, getParser(tiddler), highlightRegExp, tiddler);
+	var t0 = new Date();
+	wikifier.subWikify(output);
+	if(tiddler && config.options.chkDisplayInstrumentation)
+		displayMessage("wikify:" + tiddler.title + " in " + (new Date() - t0) + " ms");
 }
 
-function wikifyStatic(source,highlightRegExp,tiddler,format)
+//# get static html for RSS and alike
+function wikifyStatic(source, highlightRegExp, tiddler, format)
 {
-	var e = createTiddlyElement(document.body,"pre");
+	if(!source) return "";
+	if(!tiddler) tiddler = new Tiddler("temp");
+	var wikifier = new Wikifier(source, getParser(tiddler, format), highlightRegExp, tiddler);
+	wikifier.isStatic = true;
+
+	var e = createTiddlyElement(document.body, "pre");
 	e.style.display = "none";
-	var html = "";
-	if(source && source != "") {
-		if(!tiddler)
-			tiddler = new Tiddler("temp");
-		var wikifier = new Wikifier(source,getParser(tiddler,format),highlightRegExp,tiddler);
-		wikifier.isStatic = true;
-		wikifier.subWikify(e);
-		html = e.innerHTML;
-		jQuery(e).remove();
-	}
+	wikifier.subWikify(e);
+	var html = e.innerHTML;
+	jQuery(e).remove();
 	return html;
```

Comment:
It seems the prettyfier follows the rule: "put the declaration near the code where it is used", which IMO makes sense for compiled programming languages like C. But it technically it doens't matter for JS and in some cases as written above imo it even worsens code quality.

Reply:
I'm OK with that